src/snowflake/snowpark/modin/plugin/_internal/io_utils.py

```diff
@@ -126,36 +129,37 @@ def get_columns_to_keep_for_usecols(
     ValueError
         If column(s) expected in `usecols` are not found in frame's columns `columns`.
     """
-
-    if not isinstance(usecols, list):
-        keep = [column for column in columns if usecols(column)]
+    _usecols = usecols
+    if callable(_usecols):
+        keep = [column for column in columns if _usecols(column)]
+    elif len(_usecols) == 0:
+        keep = []
     else:
-        if not usecols:
-            keep = []
+        if isinstance(_usecols, native_pandas.core.series.Series):
+            _usecols = _usecols.values
+
+        if isinstance(_usecols[0], str):  # type: ignore
+            invalid_columns = [column for column in _usecols if column not in columns]  # type: ignore
+            if invalid_columns:
+                raise ValueError(
+                    f"'usecols' do not match columns, columns expected but not found: {invalid_columns}"
+                )
         else:
-            if isinstance(usecols[0], str):
-                invalid_columns = [
-                    column for column in usecols if column not in columns
-                ]
-                if invalid_columns:
-                    raise ValueError(
-                        f"'usecols' do not match columns, columns expected but not found: {invalid_columns}"
-                    )
-            else:
-                invalid_columns = [
-                    column for column in usecols if column < 0 or column >= len(columns)  # type: ignore[operator]
-                ]
-                if invalid_columns:
-                    raise ValueError(
-                        f"'usecols' do not match columns, columns expected but not found: {invalid_columns}"
-                    )
-
-                # Turn index references to pandas labels.
-                usecols = [columns[column] for column in usecols]
-
-            l1, l2 = (
-                (usecols, columns) if maintain_usecols_order else (columns, usecols)
-            )
-            keep = [column for column in l1 if column in l2]
-
+            if not all(isinstance(c, int) or isinstance(c, np.int64) for c in _usecols):  # type: ignore
```

Comment:
```suggestion
            if not all(isinstance(c, (int, np.int64)) for c in _usecols):  # type: ignore
```

optional nit

Reply:
This one I did not take. I can see how it might be useful, but it is a little harder for me (personally) to read.